File: app/src/main/java/com/haishinkit/util/ByteBufferUtils.java
package com.haishinkit.util;

import java.nio.ByteBuffer;

public final class ByteBufferUtils {
    private ByteBufferUtils() {
    }

    public static String toHexString(final  ByteBuffer buffer) {
        StringBuilder builder = new StringBuilder();
        ByteBuffer slice = buffer.slice();
        for (int i = 0; i < slice.limit(); ++i) {
            builder.append(String.format("0x%02x,", slice.get() & 0xff));
        }
        return builder.toString();
    }
}

File: haishinkit/src/main/java/com/haishinkit/util/ByteBufferUtils.kt
package com.haishinkit.util

import org.apache.commons.lang3.StringUtils

import java.nio.ByteBuffer

object ByteBufferUtils {

    fun toHexString(buffer: ByteBuffer?): String {
        if (buffer == null) {
            return StringUtils.EMPTY
        }
        val builder = StringBuilder()
        val slice = buffer.slice()
        for (i in 0..slice.limit() - 1) {
            builder.append(String.format("0x%02x,", slice.get().toInt() and 0xff))
        }
        return builder.toString()
    }
}
